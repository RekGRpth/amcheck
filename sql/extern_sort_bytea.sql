CREATE TEMP TABLE bytea_test (i bytea);

-- Seed random number generator to make outcome deterministic.
SELECT setseed(0.5);

INSERT INTO bytea_test SELECT decode(repeat(md5(random()::text), (random() * 100)::int4), 'hex')
      FROM generate_series(0, 100000);

SET maintenance_work_mem = '1MB';
CREATE INDEX idx_bytea ON bytea_test(i);
SELECT bt_index_parent_check('idx_bytea');
SET maintenance_work_mem = '3MB';
REINDEX INDEX idx_bytea;
SELECT bt_index_parent_check('idx_bytea');
SET maintenance_work_mem = '5MB';
REINDEX INDEX idx_bytea;
SELECT bt_index_parent_check('idx_bytea');
SET maintenance_work_mem = '20MB';
REINDEX INDEX idx_bytea;
SELECT bt_index_parent_check('idx_bytea');

-- drop the temporary table
DROP TABLE bytea_test;
