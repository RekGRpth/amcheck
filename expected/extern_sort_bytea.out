CREATE TEMP TABLE bytea_test (i bytea);
-- Seed random number generator to make outcome deterministic.
SELECT setseed(0.5);
 setseed 
---------
 
(1 row)

INSERT INTO bytea_test SELECT decode(repeat(md5(random()::text), (random() * 100)::int4), 'hex')
      FROM generate_series(0, 100000);
SET maintenance_work_mem = '1MB';
CREATE INDEX idx_bytea ON bytea_test(i);
SELECT bt_index_parent_check('idx_bytea');
 bt_index_parent_check 
-----------------------
 
(1 row)

SET maintenance_work_mem = '3MB';
REINDEX INDEX idx_bytea;
SELECT bt_index_parent_check('idx_bytea');
 bt_index_parent_check 
-----------------------
 
(1 row)

SET maintenance_work_mem = '5MB';
REINDEX INDEX idx_bytea;
SELECT bt_index_parent_check('idx_bytea');
 bt_index_parent_check 
-----------------------
 
(1 row)

SET maintenance_work_mem = '20MB';
REINDEX INDEX idx_bytea;
SELECT bt_index_parent_check('idx_bytea');
 bt_index_parent_check 
-----------------------
 
(1 row)

-- drop the temporary table
DROP TABLE bytea_test;
